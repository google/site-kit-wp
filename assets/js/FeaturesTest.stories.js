import { useSelect } from 'googlesitekit-data';
import { WithTestRegistry } from '../../tests/js/utils';
import { CORE_SITE } from './googlesitekit/datastore/site/constants';
import { useFeature } from './hooks/useFeature';

function FeaturesTest() {
	const isFooEnabled = useFeature( 'foo' );

	return <p>Feature foo is { isFooEnabled ? 'enabled' : 'disabled' }</p>;
}

function WasFeatureEnabledInCallback() {
	const wasFooEnabled = useSelect( ( select ) =>
		select( CORE_SITE ).wasFeatureEnabled( 'foo' )
	);

	return (
		<p>
			Was feature foo enabled during callback?&nbsp;
			{ wasFooEnabled ? 'yes' : 'no' }
		</p>
	);
}

export function WithDefaultDecorator() {
	return (
		<div>
			<h1>With Default Decorator</h1>
			<FeaturesTest />
		</div>
	);
}
WithDefaultDecorator.parameters = {
	features: [ 'foo' ],
};

export function WithTestRegistryInStory() {
	return (
		<div>
			<h1>With Test Registry In Story</h1>
			<WithTestRegistry
				features={ [ 'foo' ] }
				callback={ ( registry ) => {
					registry.dispatch( CORE_SITE ).checkFeatureEnabled( 'foo' );
				} }
			>
				<FeaturesTest />
				<WasFeatureEnabledInCallback />
			</WithTestRegistry>
		</div>
	);
}

export default {
	title: 'FeaturesTest',
	component: FeaturesTest,
};
