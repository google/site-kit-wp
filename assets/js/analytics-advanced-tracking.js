/**
 * Analytics advanced tracking script to be inserted into the frontend via PHP.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// This file should not use any dependencies because it is used in the frontend.

/**
 * Internal dependencies
 */
import setUpAdvancedTracking from './analytics-advanced-tracking/set-up-advanced-tracking';

/**
 * Sends a tracking event to Analytics via gtag.
 *
 * @since 1.18.0
 *
 * @param {string} action   Event action / event name.
 * @param {Object} metadata Additional event metadata to send, or `null`.
 */
function sendEvent( action, metadata ) {
	window.gtag( 'event', action, metadata || undefined ); // eslint-disable-line no-restricted-globals
}

const events = window._googlesitekitAnalyticsTrackingData || []; // eslint-disable-line no-restricted-globals
if ( Array.isArray( events ) ) {
	setUpAdvancedTracking( events, sendEvent );
	// eslint-disable-next-line no-console
	console.log( 'Hello QA testing' );
}
