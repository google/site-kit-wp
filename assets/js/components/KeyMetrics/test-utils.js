/**
 * Key Metrics test utility functions.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { CORE_WIDGETS } from '../../googlesitekit/widgets/datastore/constants';
import { AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY } from '../../googlesitekit/widgets/default-areas';
import { CONTEXT_MAIN_DASHBOARD_KEY_METRICS } from '../../googlesitekit/widgets/default-contexts';

/**
 * Provides key metric widgets registration data to the given registry.
 *
 * This is a temporary utility until a solidified solution is implemented to effortlessly provide
 * widget registrations to the registry.
 *
 * @since 1.105.0
 * @see {@link https://github.com/google/site-kit-wp/issues/7264} Initiative to implement a utility to provide widget registrations.
 *
 * @param {Object} registry The registry to set up.
 * @param {Object} widgets  Object containing options mapped to widget slugs.
 */
export function provideKeyMetricsWidgetRegistrations( registry, widgets ) {
	registry
		.dispatch( CORE_WIDGETS )
		.registerWidgetArea( AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY, {
			title: 'Key metrics',
		} );

	registry
		.dispatch( CORE_WIDGETS )
		.assignWidgetArea(
			AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY,
			CONTEXT_MAIN_DASHBOARD_KEY_METRICS
		);

	Object.keys( widgets ).forEach( ( slug ) => {
		registry.dispatch( CORE_WIDGETS ).registerWidget( slug, {
			Component() {
				return <div>Hello test.</div>;
			},
			...widgets[ slug ],
		} );

		registry
			.dispatch( CORE_WIDGETS )
			.assignWidget( slug, AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY );
	} );
}
