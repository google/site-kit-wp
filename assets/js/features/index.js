/**
 * Feature flags.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const enabledFeatures = new Set(
	global?._googlesitekitBaseData?.enabledFeatures || []
);

/**
 * Returns true if a feature is enabled; false otherwise.
 *
 * @since 1.25.0
 * @since 1.33.0 Changed _enabledFeatures argument to be a `Set` instead of `Array`.
 *
 * @param {string} feature            The name of the feature to check.
 * @param {Set}    [_enabledFeatures] Optional. The set of enabled features. Uses `enabledFeatures` set by the server in a global JS variable, by default.
 * @return {boolean} `true` if a feature is enabled; `false` otherwise.
 */
export function isFeatureEnabled(
	feature,
	_enabledFeatures = enabledFeatures
) {
	if ( ! ( _enabledFeatures instanceof Set ) ) {
		return false;
	}
	return _enabledFeatures.has( feature );
}
