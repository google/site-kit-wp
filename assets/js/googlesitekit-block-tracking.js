import { watchBlocks } from './googlesitekit/block-tracking/watch-blocks';

const blocksToTrack = [
	'google-site-kit/rrm-subscribe-with-google',
	'google-site-kit/rrm-contribute-with-google',
];

watchBlocks( blocksToTrack );
