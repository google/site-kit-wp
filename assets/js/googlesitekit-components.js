/**
 * Public components entrypoint.
 *
 * Site Kit by Google, Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import Components from './googlesitekit/components-gm2';

if ( typeof global.googlesitekit === 'undefined' ) {
	global.googlesitekit = {};
}

global.googlesitekit.components = Components;

export const {
	Button,
	Checkbox,
	Chip,
	CircularProgress,
	Dialog,
	DialogTitle,
	DialogContent,
	DialogFooter,
	Menu,
	Option,
	ProgressBar,
	Radio,
	Select,
	SpinnerButton,
	Switch,
	Tab,
	TabBar,
	TextField,
	HelperText,
	Tooltip,
} = Components;
