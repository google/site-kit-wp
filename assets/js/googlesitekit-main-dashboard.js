/**
 * Dashboard component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import domReady from '@wordpress/dom-ready';
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { clearCache } from './googlesitekit/api/cache';
import Root from './components/Root';
import {
	VIEW_CONTEXT_MAIN_DASHBOARD,
	VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY,
	VIEW_CONTEXT_MODULE_SETUP,
} from './googlesitekit/constants';
import DashboardEntryPoint from './components/DashboardEntryPoint';

// Initialize the app once the DOM is ready.
domReady( async () => {
	if ( global._googlesitekitLegacyData.admin.resetSession ) {
		await clearCache();
	}

	const renderTarget = document.getElementById(
		'js-googlesitekit-main-dashboard'
	);

	if ( renderTarget ) {
		const { setupModuleSlug, viewOnly } = renderTarget.dataset;

		let viewContext = VIEW_CONTEXT_MODULE_SETUP;
		if ( ! setupModuleSlug ) {
			viewContext = viewOnly
				? VIEW_CONTEXT_MAIN_DASHBOARD_VIEW_ONLY
				: VIEW_CONTEXT_MAIN_DASHBOARD;
		}

		render(
			<Root viewContext={ viewContext }>
				<DashboardEntryPoint setupModuleSlug={ setupModuleSlug } />
			</Root>,
			renderTarget
		);
	}
} );
