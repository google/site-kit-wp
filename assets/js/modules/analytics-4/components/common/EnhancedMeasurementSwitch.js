/**
 * EnhancedMeasurementSwitch component.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { useMount } from 'react-use';

/**
 * WordPress dependencies
 */
import { createInterpolateElement, useCallback } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect, useDispatch } from 'googlesitekit-data';
import { ProgressBar, Switch } from 'googlesitekit-components';
import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
import {
	ENHANCED_MEASUREMENT_ENABLED,
	ENHANCED_MEASUREMENT_FORM,
	ENHANCED_MEASUREMENT_SHOULD_DISMISS_ACTIVATION_BANNER,
} from '../../datastore/constants';
import SupportLink from '../../../../components/SupportLink';
import { trackEvent } from '../../../../util';
import useViewContext from '../../../../hooks/useViewContext';
import Tick from '../../../../../svg/icons/tick.svg';

export default function EnhancedMeasurementSwitch( {
	className,
	onClick,
	disabled = false,
	loading = false,
	formName = ENHANCED_MEASUREMENT_FORM,
	isEnhancedMeasurementAlreadyEnabled = false,
	showTick = false,
} ) {
	const isEnhancedMeasurementEnabled = useSelect( ( select ) =>
		select( CORE_FORMS ).getValue( formName, ENHANCED_MEASUREMENT_ENABLED )
	);

	const viewContext = useViewContext();
	const { setValues } = useDispatch( CORE_FORMS );

	const handleClick = useCallback( () => {
		setValues( formName, {
			[ ENHANCED_MEASUREMENT_ENABLED ]: ! isEnhancedMeasurementEnabled,
		} );

		trackEvent(
			`${ viewContext }_analytics`,
			// If the current status of enhanced measurement is enabled,
			// then it means that we disabled it, otherwise enabled it.
			isEnhancedMeasurementEnabled
				? 'deactivate_enhanced_measurement'
				: 'activate_enhanced_measurement'
		);

		onClick?.();
	}, [
		formName,
		isEnhancedMeasurementEnabled,
		onClick,
		setValues,
		viewContext,
	] );

	useMount( () => {
		// Ensure the Enhanced Measurement activation banner won't be shown if we've updated the setting
		// via the switch.
		setValues( ENHANCED_MEASUREMENT_FORM, {
			[ ENHANCED_MEASUREMENT_SHOULD_DISMISS_ACTIVATION_BANNER ]: true,
		} );
	} );

	return (
		<div
			className={ classnames(
				'googlesitekit-analytics-enable-enhanced-measurement',
				className,
				{
					'googlesitekit-analytics-enable-enhanced-measurement--loading':
						loading,
				}
			) }
		>
			{ loading && (
				<ProgressBar
					small
					className="googlesitekit-analytics-enable-enhanced-measurement__progress--settings-edit"
				/>
			) }
			{ ! loading && isEnhancedMeasurementAlreadyEnabled && (
				<div className="googlesitekit-analytics-enable-enhanced-measurement__already-enabled-label">
					{ showTick && (
						<div className="googlesitekit-analytics-enable-enhanced-measurement__already-enabled-tick">
							<Tick />
						</div>
					) }
					{ __(
						'Enhanced measurement is enabled for this web data stream',
						'google-site-kit'
					) }
				</div>
			) }
			{ ! loading && ! isEnhancedMeasurementAlreadyEnabled && (
				<Switch
					label={ __(
						'Enable enhanced measurement',
						'google-site-kit'
					) }
					checked={ isEnhancedMeasurementEnabled }
					disabled={ disabled }
					onClick={ handleClick }
					hideLabel={ false }
				/>
			) }
			<p className="googlesitekit-module-settings-group__helper-text">
				{ createInterpolateElement(
					__(
						'This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). <a>Learn more</a>',
						'google-site-kit'
					),
					{
						a: (
							<SupportLink
								path="/analytics/answer/9216061"
								external
							/>
						),
					}
				) }
			</p>
		</div>
	);
}

EnhancedMeasurementSwitch.propTypes = {
	onClick: PropTypes.func,
	disabled: PropTypes.bool,
	loading: PropTypes.bool,
	isEnhancedMeasurementAlreadyEnabled: PropTypes.bool,
	showTick: PropTypes.bool,
};
