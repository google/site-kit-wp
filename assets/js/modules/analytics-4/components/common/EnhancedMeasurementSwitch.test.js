/**
 * EnhancedMeasurementSwitch tests.
 *
 * Site Kit by Google, Copyright 2023 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { render } from '../../../../../../tests/js/test-utils';
import EnhancedMeasurementSwitch from './EnhancedMeasurementSwitch';

describe( 'EnhancedMeasurementSwitch', () => {
	it( 'should render correctly in the default state', () => {
		const { container, getByLabelText, queryByText } = render(
			<EnhancedMeasurementSwitch />
		);

		expect( container ).toMatchSnapshot();

		expect(
			getByLabelText( 'Enable enhanced measurement' )
		).toBeInTheDocument();

		expect(
			queryByText(
				'Enhanced measurement is enabled for this web data stream'
			)
		).not.toBeInTheDocument();
	} );

	it( 'should render correctly in the loading state', () => {
		const { container, getByRole } = render(
			<EnhancedMeasurementSwitch loading />
		);

		expect( container ).toMatchSnapshot();

		expect( getByRole( 'progressbar' ) ).toBeInTheDocument();
	} );

	it( 'should render correctly in the disabled state', () => {
		const { container, getByLabelText } = render(
			<EnhancedMeasurementSwitch disabled />
		);

		expect( container ).toMatchSnapshot();

		expect(
			getByLabelText( 'Enable enhanced measurement' )
		).toBeDisabled();
	} );

	it( 'should render correctly in the already enabled state', () => {
		const { container, queryByLabelText, getByText } = render(
			<EnhancedMeasurementSwitch isEnhancedMeasurementAlreadyEnabled />
		);

		expect( container ).toMatchSnapshot();

		expect(
			getByText(
				'Enhanced measurement is enabled for this web data stream'
			)
		).toBeInTheDocument();

		expect(
			queryByLabelText( 'Enable enhanced measurement' )
		).not.toBeInTheDocument();
	} );

	it( 'should toggle the switch on click', () => {
		const { getByLabelText } = render( <EnhancedMeasurementSwitch /> );

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		expect( switchControl ).not.toBeChecked();

		switchControl.click();

		expect( switchControl ).toBeChecked();

		// Give it another click to verify it can be toggled off.
		switchControl.click();

		expect( switchControl ).not.toBeChecked();
	} );

	it( 'should invoke the onClick callback when clicked', () => {
		const onClick = vi.fn();

		const { getByLabelText } = render(
			<EnhancedMeasurementSwitch onClick={ onClick } />
		);

		const switchControl = getByLabelText( 'Enable enhanced measurement' );

		switchControl.click();

		expect( onClick ).toHaveBeenCalledTimes( 1 );
	} );
} );
