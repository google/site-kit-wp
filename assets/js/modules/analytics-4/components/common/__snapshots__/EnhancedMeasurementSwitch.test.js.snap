// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnhancedMeasurementSwitch should render correctly in the already enabled state 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      class="googlesitekit-analytics-enable-enhanced-measurement__already-enabled-label"
    >
      Enhanced measurement is enabled for this web data stream
    </div>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`EnhancedMeasurementSwitch should render correctly in the default state 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="false"
      class="mdc-switch"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            id="googlesitekit-switch-0"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-0"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`EnhancedMeasurementSwitch should render correctly in the disabled state 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement"
  >
    <div
      aria-checked="false"
      class="mdc-switch mdc-switch--disabled"
      role="switch"
      tabindex="0"
    >
      <div
        class="mdc-switch__track"
      >
         
      </div>
      <div
        class="mdc-switch__thumb-underlay"
      >
        <div
          class="mdc-switch__thumb"
        >
          <input
            class="mdc-switch__native-control"
            disabled=""
            id="googlesitekit-switch-1"
            readonly=""
            role="switch"
            type="checkbox"
          />
        </div>
      </div>
    </div>
    <label
      class=""
      for="googlesitekit-switch-1"
    >
      Enable enhanced measurement
    </label>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;

exports[`EnhancedMeasurementSwitch should render correctly in the loading state 1`] = `
<div>
  <div
    class="googlesitekit-analytics-enable-enhanced-measurement googlesitekit-analytics-enable-enhanced-measurement--loading"
  >
    <div
      class="mdc-linear-progress googlesitekit-analytics-enable-enhanced-measurement__progress--settings-edit mdc-linear-progress--indeterminate mdc-linear-progress--small"
      role="progressbar"
    >
      <div
        class="mdc-linear-progress__buffering-dots"
      />
      <div
        class="mdc-linear-progress__buffer"
      />
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
      <div
        class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
      >
        <span
          class="mdc-linear-progress__bar-inner"
        />
      </div>
    </div>
    <p
      class="googlesitekit-module-settings-group__helper-text"
    >
      This allows you to measure interactions with your content (e.g. file downloads, form completions, video views). 
      <a
        aria-label="Learn more (opens in a new tab)"
        class="googlesitekit-cta-link"
        href="https://support.google.com/analytics/answer/9216061?hl=en-US"
        rel="noopener noreferrer"
        target="_blank"
      >
        <span
          class="googlesitekit-cta-link__contents"
        >
          Learn more
        </span>
        <span
          class="googlesitekit-icon-wrapper"
          style="margin-left: 5px;"
        >
          <svg />
        </span>
      </a>
    </p>
  </div>
</div>
`;
